dlp-terraform/ecs/ecs_training_service.tf

```diff
@@ -0,0 +1,212 @@
+          "awslogs-region" : "us-east-1",
+          "awslogs-group" : aws_cloudwatch_log_group.training.name,
+          "awslogs-stream-prefix" : "ecs"
+        }
+      }
+    }
+  ])
+}
+
+# --- ECS Service ---
+resource "aws_ecs_service" "training" {
+  name            = "training"
+  cluster         = aws_ecs_cluster.main.id
+  task_definition = aws_ecs_task_definition.training.arn
+  desired_count   = 2
+
+  capacity_provider_strategy {
+    capacity_provider = aws_ecs_capacity_provider.training.name
+    base              = 1
+    weight            = 100
+  }
+
+  ordered_placement_strategy {
+    type  = "spread"
+    field = "attribute:ecs.availability-zone"
+  }
+
+  lifecycle {
+    ignore_changes = [desired_count]
+  }
+
+  # load_balancer {
+  #   target_group_arn = aws_lb_target_group.app.arn
+  #   container_name = "training"
+  #   container_port = 8000
+  # }
+
+  # depends_on = [aws_lb_target_group.app]
+}
+
+# --- ECS Service Auto Scaling ---
+resource "aws_appautoscaling_target" "training_ecs_target" {
+  service_namespace = "ecs"
+  scalable_dimension = "ecs:service:DesiredCount"
+  resource_id = "service/${aws_ecs_cluster.main.name}/${aws_ecs_service.training.name}"
+  min_capacity = 0
+  max_capacity = 2
+}
+
+resource "aws_appautoscaling_policy" "training_ecs_target_cpu" {
+  name = "training-application-scaling-policy-cpu"
+  policy_type = "TargetTrackingScaling"
+  service_namespace = aws_appautoscaling_target.training_ecs_target.service_namespace
+  resource_id = aws_appautoscaling_target.training_ecs_target.resource_id
+  scalable_dimension = aws_appautoscaling_target.training_ecs_target.scalable_dimension
+
+  target_tracking_scaling_policy_configuration {
+    predefined_metric_specification {
+      predefined_metric_type = "ECSServiceAverageCPUUtilization"
+    }
+
+    target_value = 80
+    scale_in_cooldown = 300
+    scale_out_cooldown = 300
+  }
+}
+
+resource "aws_appautoscaling_policy" "training_ecs_target_memory" {
+  name = "training-application-scaling-policy-memory"
+  policy_type = "TargetTrackingScaling"
+  service_namespace = aws_appautoscaling_target.training_ecs_target.service_namespace
+  resource_id = aws_appautoscaling_target.training_ecs_target.resource_id
+  scalable_dimension = aws_appautoscaling_target.training_ecs_target.scalable_dimension
+
+  target_tracking_scaling_policy_configuration {
+    predefined_metric_specification {
+      predefined_metric_type = "ECSServiceAverageMemoryUtilization"
+    }
+
+    target_value = 80
+    scale_in_cooldown = 300
+    scale_out_cooldown = 300
+  }
+}
+
+# --- ECS Training Security Group ---
+resource "aws_security_group" "ecs_training_sg" {
+  name_prefix = "backend-ecs-training-sg-"
+  vpc_id      = aws_vpc.main.id
+}
+
+resource "aws_vpc_security_group_ingress_rule" "ecs_training_sg_ingress" {
+  security_group_id = aws_security_group.ecs_training_sg.id
+
+  ip_protocol    = "-1"
+  # cidr_blocks = [aws_vpc.main.cidr_block]
```

Comment:
commented out code?

Reply:
do we need this?

Reply:
I'm not sure yet, will be resolved in the next ticket most likely

Reply:
ok.